src/library_gl.js

```diff
@@ -1023,6 +1023,13 @@ var LibraryGL = {
 
   glCompressedTexImage2D__sig: 'viiiiiiii',
   glCompressedTexImage2D: function(target, level, internalFormat, width, height, border, imageSize, data) {
+#if USE_WEBGL2
+    if (GL.currentContext.version >= 2) { // WebGL 2 provides new garbage-free entry points to call to WebGL. Use those always when possible.
+      GLctx['compressedTexImage2D'](target, level, internalFormat, width, height, border, HEAP8, data, imageSize);
```

Comment:
it likely doesn't matter, but we access `HEAPU8` below in this method, so maybe the same here?